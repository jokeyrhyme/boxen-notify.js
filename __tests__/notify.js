'use strict'

const EventEmitter = require('events').EventEmitter

const stripAnsi = require('strip-ansi')

const notify = require('../index.js').notify

function clearRequire (id) {
  delete require.cache[id]
}

// mock console.error
let oldConsoleError
afterEach(() => {
  console.error = oldConsoleError
})

// mock interactive terminal and non-`npm run`-environment
let processEnvBefore
let isTTYBefore
beforeEach(function () {
  ['is-npm'].forEach(clearRequire)
  processEnvBefore = JSON.stringify(process.env)
  isTTYBefore = process.stdout.isTTY
  ;['npm_config_username', 'npm_package_name', 'npm_config_heading'].forEach(function (name) {
    delete process.env[name]
  })
  process.stdout.isTTY = true
})
afterEach(function () {
  ['is-npm'].forEach(clearRequire)
  process.env = JSON.parse(processEnvBefore)
  process.stdout.isTTY = isTTYBefore
})

test('notify({ message: "hello, world!" })', () => {
  console.error = jest.fn()
  notify({ message: 'hello, world!' })
  expect(console.error).toHaveBeenCalledTimes(1)

  const uncoloredOutput = stripAnsi(console.error.mock.calls[0][0])
  expect(uncoloredOutput).toMatchSnapshot()
})

// mock process.on
let processOnBefore = process.on
let emit
beforeEach(() => {
  const emitter = new EventEmitter()
  process.on = emitter.on.bind(emitter)
  emit = emitter.emit.bind(emitter)
})
afterEach(() => {
  process.on = processOnBefore
  emit = null
})

test('notify({ message: "hello, world!", defer: true })', () => {
  console.error = jest.fn()
  notify({ message: 'hello, world!', defer: true })
  expect(console.error).not.toBeCalled()

  emit('exit')
  expect(console.error).toHaveBeenCalledTimes(1)

  const uncoloredOutput = stripAnsi(console.error.mock.calls[0][0])
  expect(uncoloredOutput).toMatchSnapshot()
})
