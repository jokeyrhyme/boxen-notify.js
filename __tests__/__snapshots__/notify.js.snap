// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notify({ message: "hello, world!" }) 1`] = `
"

   ╭───────────────────╮
   │                   │
   │   hello, world!   │
   │                   │
   ╰───────────────────╯
"
`;
